test/integration/ethereum/stakeWiseReinvestmentExtensionIntegration.spec.ts

```diff
@@ -0,0 +1,243 @@
+
+      sETH2 = (await ethers.getContractAt("IERC20", addresses.tokens.sETH2)) as IERC20;
+
+      await deployStakeWiseReinvestmentExtension();
+
+      await extension.connect(operator).updateCallerStatus([await operator.getAddress()], [true]);
+
+      // Modules need to first be added through the manager before the extension can initialize.
+      // TODO (Richard): these lines can be removed once modeuls get added to the manager and blockNumber is advanced
+      await manager.connect(operator).addModule(airdropModule.address);
+      await manager.connect(operator).addModule(tradeModule.address);
+
+      await manager.connect(operator).authorizeInitialization();
+      await manager.connect(operator).addExtension(extension.address);
+
+      // Fund owner wallet from whale
+      await sETH2.connect(whale).transfer(owner.address, ether(1));
+
+      const debtIssuanceModule = (await ethers.getContractAt(
+        "IDebtIssuanceModule",
+        addresses.setFork.debtIssuanceModuleV2,
+      )) as IDebtIssuanceModule;
+
+      await sETH2.connect(owner.wallet).approve(debtIssuanceModule.address, MAX_UINT_256);
+      await rETH2.connect(owner.wallet).approve(debtIssuanceModule.address, MAX_UINT_256);
+
+      // Issue 1 wsETH2 for owner
+      await debtIssuanceModule
+        .connect(owner.wallet)
+        .issue(setToken.address, ether(1), owner.address);
+    });
+
+    afterEach(async () => {
+      await network.provider.send("evm_revert", [snapshotId]);
+    });
+
+    async function deployStakeWiseReinvestmentExtension() {
+      extension = await deployer.extensions.deployStakeWiseReinvestmentExtension(
+        manager.address,
+        airdropModule.address,
+        tradeModule.address,
+        {
+          exchangeName: "UniswapV3ExchangeAdapterV2",
+          exchangeCallData:
+            // From generateDataParams on UniswapV3ExchangeAdapterV2
+            "0x20bc832ca081b91433ff6c17f85701b6e92486c50001f4fe2e637202056d30016725477c5da089ab0a043a01",
+        },
+      );
+    }
+
+    describe("#constructor", () => {
+      it("should set the correct manager address", async () => {
+        expect(await extension.manager()).to.eq(manager.address);
+      });
+
+      it("should set the correct airdropModule address", async () => {
+        expect(await extension.airdropModule()).to.eq(airdropModule.address);
+      });
+
+      it("should set the correct tradeModule address", async () => {
+        expect(await extension.tradeModule()).to.eq(tradeModule.address);
+      });
+
+      it("should set the correct executionSettings", async () => {
+        expect(await extension.settings()).to.deep.eq([
+          "UniswapV3ExchangeAdapterV2",
+          "0x20bc832ca081b91433ff6c17f85701b6e92486c50001f4fe2e637202056d30016725477c5da089ab0a043a01",
+        ]);
+      });
+    });
+
+    describe("#initialize", () => {
+      async function subject() {
```

Comment:
For completeness I'd add some tests that assert that unauthorized callers cannot call protected methods.